wiki/osu!wiki_contribution_guide/en.md

```diff
@@ -1,178 +1,895 @@
+
+Pick the branch you want to pull from and a page refresh will occur.
+
+Fill up the PR information (you can just write anything since it is your own repo), and press the green "Create pull request".
+
+You will be directed to your freshly-created PR page:
+
+![Preparing to merge](./img/Contribute_syncmerge1.jpg "Preparing to merge")
+
+Scroll down to the lower part of the page.
+Always set your green "Merge pull request" button to "Create a merge commit", which is set by default.
+
+Press the green "Merge pull request" button.
+
+![Setting the merge message](./img/Contribute_syncmerge2.jpg "Setting the merge message")
+
+Click on the green "Confirm merge" button and the branch will be updated successfully to the latest version.
+
+To update your other branches, do the same thing in this section in a new PR but with different branch this time.
+
+#### Merge conflicts
+
+[Read this GitHub Help for an overview about merge conflicts][GH Help webmerge conflicts]
+
+[Read this GitHub Help on resolving merge conflicts in GitHub itself][GH Help solve webmerge conflicts]
+
+### Through your Local Machine
+
+#### Branches cleanups
+
+_Note: It is highly suggested to do branch cleanups locally.
+It is best not do any web branch cleaning in GitHub if you have a local copy._
+
+```prolog
+git checkout master
+git branch -d <branchname>
+git push origin :<branchname>
+```
+
+- The first command set the branch to master.
+  This will allow you to delete the other branches.
+- The second command will tell Git to delete the branch.
+  If Git refuse to delete it for a reason (usually the branch was not updated) but you do not want the branch anyway, use the force delete variant by replacing the soft local delete attribute, ``-d``, to force local delete attribute, ``-D``.
+- The third command will update the forked remote repo to also remove the branch (make sure you are connected to the Internet).
+  The colon, ``:``, will tells Git to delete this ``<branchname>`` branch in the ``origin`` nickname (which is your forked remote repo).
+
+#### Branch sync
+
+Do a checkout to the branch you want to sync.
+
+```prolog
+git checkout <branchname>
+```
+
+There are two ways of updating it:
+
+- The first way provides a open choice (will not be explained in this guide) to do a forensic check on what was changed before doing any merging.
+- The second way is best used if you know that there will not be any merge conflicts, does not care about what was changed, and lazy to type two commands.
+
+Both commands will provide the same outcome, and will complain about the merge conflict(s) if found.
+
+##### Fetch + Merge
+
+```prolog
+git fetch https://github.com/ppy/osu-wiki.git
+git merge https://github.com/ppy/osu-wiki.git/master
+git push origin <branchname>
+```
+
+or, if you nicknamed ``https://github.com/ppy/osu-wiki.git`` to ``upstream`` for example, then the below will work too:
+
+```prolog
+git fetch upstream
+git merge upstream/master
```

Comment:
Why?

It is updating the own master branch to the latest master version available.

It is probably relevant to team osu! merging to official master branch, not updating.